guides/keras_nlp/upload.py

```diff
@@ -0,0 +1,238 @@
+In this section, we explain how to create a `Task`, fine-tune and upload it to a model hub.
+"""
+
+"""
+## Load Model
+
+If you want to build a Causal LM based on a base model, simply call `keras_nlp.models.Classifier.from_preset`
+and pass a built-in preset identifier.
+"""
+
+causal_lm = keras_nlp.models.CausalLM.from_preset("gpt2_base_en")
+
+
+"""
+## Fine-tune Model
+
+After loading the model, you can call `.fit()` on the model to fine-tune it.
+Here, we fine-tune the model on the IMDB reviews which makes the model movie domain-specific.
+"""
+
+# Drop labels and keep the review text only for the Causal LM.
+imdb_train_reviews = imdb_train.map(lambda x, y: x)
+
+# Fine-tune the Causal LM.
+causal_lm.fit(imdb_train_reviews)
+
+"""
+## Save the Model Locally
+
+To upload a model, you need to first save the model locally using `save_to_preset`.
+"""
+
+preset_dir = "./finetuned_gpt2"
+causal_lm.save_to_preset(preset_dir)
+
+"""
+Let's see what are the files what are the saved files.
+"""
+
+os.listdir(preset_dir)
+
+"""
+### Load a Locally Saved Model
+
+A model that is saved to a local preset, can be loaded using `from_preset`.
+"""
+
+causal_lm = keras_nlp.models.CausalLM.from_preset(preset_dir)
+
+"""
+You can also load the `Backbone` and `Tokenizer` objects from this preset directory.
+"""
+
+backbone = keras_nlp.models.Backbone(preset_dir)
+tokenizer = keras_nlp.models.Tokenizer(preset_dir)
+
+"""
+## Upload the Model to a Model Hub
+
+When the model is saved to a local directory `preset_dir`, this directory can be seamlessly
+uploaded to a model hub such as Kaggle or Hugging Face programmatically.
+"""
+"""
+### Upload to Kaggle Models
+"""
+
+"""
+To upload a model to Kaggle Models, first, we need to authenticate with Kaggle.
+This can by one of the followings:
+1. Set environment variables `KAGGLE_USERNAME` and `KAGGLE_KEY`.
+2. Provide a local `~/.kaggle/kaggle.json`.
+3. Call `kagglehub.login()`.
+
+Let's make sure we are logged in before coninuing.
+"""
+
+import kagglehub
+
+if "KAGGLE_USERNAME" not in os.environ or "KAGGLE_KEY" not in os.environ:
+    kagglehub.login()
+
+
+"""
+Run this command to upload the model that is save in `preset_dir` to Kaggle Models:
+"""
+
+kaggle_username = ""  # TODO: assing username.
+kaggle_uri = f"kaggle://{kaggle_username}/gpt2/keras/finetuned_gpt2"
+keras_nlp.upload_preset(kaggle_uri, preset_dir)
+
+"""
+### Upload to Hugging Face Hub
+"""
+
+"""
+To upload a model to Hugging Face Hub, first, we need to authenticate with Hugging Face.
+This can by one of the followings:
+1. Set environment variables `HF_USERNAME` and `HF_TOKEN`.
+2. Call `huggingface_hub.notebook_login()`.
+
+Let's make sure we are logged in before coninuing.
+"""
+
+import huggingface_hub
+
+if "HF_USERNAME" not in os.environ or "HF_TOKEN" not in os.environ:
+    huggingface_hub.notebook_login()
+
+"""
+Run this command to upload the model that is save in `preset_dir` to Hugging Face Hub:
```

Comment:
Somewhere in all of this we should have a little callout on differences in huggingface/kaggle upload URIs.

E.g. kaggle include the model name and framework name as segments, huggingface does not.

Reply:
Good idea! I added the format of the URI in both cases.